`bun@1.1.39` install is not correctly like `1.1.38`

### What version of Bun is running?

1.1.39

### What platform is your computer?

Linux

### What steps can reproduce the bug?

```
git clone https://github.com/lobehub/lobe-chat

bun i

npm run test-app 
```

### What is the expected behavior?

as `bun@1.1.38`, it should pass the all tests like this CI: https://github.com/lobehub/lobe-chat/actions/runs/12367455614/job/34515786344

<img width="1701" alt="Image" src="https://github.com/user-attachments/assets/30f7d7ac-e62e-426c-b06e-577bdb96c14d" />



### What do you see instead?

when use `bun@1.1.39`, the test failed like this CI: https://github.com/lobehub/lobe-chat/actions/runs/12373981587/job/34535476144

<img width="1696" alt="Image" src="https://github.com/user-attachments/assets/3c3c21c7-418e-4582-934d-8eab8ef3e58f" />

error log: 
```
FAIL  src/services/__tests__/tool.test.ts > ToolService > getToolManifest > support OpenAPI manifest > should get plugin manifest
TypeError: openAPIInvalid
 ❯ ToolService.getToolManifest src/utils/toolManifest.ts:119:13
    117|       data.settings = await convertor.convertAuthToSettingsSchema(data…
    118|     } catch (error) {
    119|       throw new TypeError('openAPIInvalid', { cause: error });
       |             ^
    120|     }
    121|   }
 ❯ src/services/__tests__/tool.test.ts:217:26

Caused by: TypeError: applicator_1.default is not a function
 ❯ Object.<anonymous> node_modules/ajv-draft-04/src/vocabulary/draft4.ts:12:26
 ❯ Object.<anonymous> node_modules/ajv-draft-04/src/index.ts:3:1
```

### Additional information

I think relative issue is: https://github.com/oven-sh/bun/issues/8201

--- a/src/install/lockfile.zig
+++ b/src/install/lockfile.zig
@@ -762,10 +762,14 @@ pub const Tree = struct {
                 const deps_buf = sorter.lockfile.buffers.dependencies.items;
                 const string_buf = sorter.lockfile.buffers.string_bytes.items;
 
-                const l_dep_name = deps_buf[l].name.slice(string_buf);
-                const r_dep_name = deps_buf[r].name.slice(string_buf);
+                const l_dep = deps_buf[l];
+                const r_dep = deps_buf[r];
 
-                return strings.order(l_dep_name, r_dep_name) == .lt;
+                return switch (l_dep.behavior.cmp(r_dep.behavior)) {
+                    .lt => true,
+                    .gt => false,
+                    .eq => strings.order(l_dep.name.slice(string_buf), r_dep.name.slice(string_buf)) == .lt,
+                };
             }
         };
 

